### DISCUSSION TITLE
Deezer ARL Support

### DISCUSSION THREAD

**Discussion by ByAhri** [TopiSenpai/LavaSrc#70] [General]
sometimes lavasrc is failing to get the proper track on deezer when it's a premium track only, is there a way to access these premium tracks by giving a premium account or something?

**Comment by topi314** [comment]
rn there is no way to access premium tracks I believe, I do plan on adding support to provide your own ARL at some point, which would also allow to play higher quality.
Just haven't gotten around to it

**Reply by ByAhri** [reply]
oh okay, thank u. I really hope you could add those things it would be so nice.

I also was wondering if there is a way to use providers dzisrc and dzsearch as optional like ytsearch isrc that can be ignored if the track doesn't have isrc, so it could look up on ytmsearch for example if it fails with deezer

**Reply by topi314** [reply]
`dzisrc` should already be ignored if the track has no `isrc`
See: https://github.com/TopiSenpai/LavaSrc/blob/master/main/src/main/java/com/github/topisenpai/lavasrc/mirror/DefaultMirroringAudioTrackResolver.java#L42

**Reply by ByAhri** [reply]
oh okay thank u!!

and again it would be so good to have that support for the own deezer thing since I have being noticing way more tracks being exclusive for premium users only than before and the player gets stuck very often.

**Reply by topi314** [reply]
I created an issue for it, https://github.com/TopiSenpai/LavaSrc/issues/71
If you really want to support it as fast as possible I accept contributions
